src/bash/file-perms.md

```diff
@@ -1 +1,214 @@
+all likelihood, these are the names of *your* user and *your* group.
+
+The systems we're using --- Linux, MacOS, or WSL on Windows --- all descend from
+an operating system called UNIX, which achieved success in part due to its
+**multi-user** support. On our personal computers, we don't generally use
+multi-user features all that frequently or explicitly, but they're baked in to
+the computer so much that, in fact, every single file is **owned by** one of
+many **users,** and, simultaneously, one of many **groups.**
+
+If you want to know what your user is called, you can always run `whoami`:
+
+```shell
+$ whoami
+atalii
+```
+
+> This is just my personal username. Yours will be different!
+
+Similarly, `groups` can tell you what groups you're a member of:
+
+```shell
+$ groups
+atalii wheel input
+```
+
+> Again, this is just the output on my machine. You'll probably be a member of
+> some different groups. You don't need to worry about what each one is for
+> right now.
+
+This is the basis for the **UNIX permission model:** Every file has three sets of
+permissions. First, the owning user can read the file, write the file, or
+execute the file. Then, a different set of permissions (one, none, or all of
+reading, writing, and executing) apply to a user who is not the owner but is in
+the owning group. Then, there's a third set of permissions for everyone else.
+
+This is actually what you see in the leftmost column of the long listing. The
+first letter (above, this is either `-` or `d`) indicates the kind of file this
+is. `-` indicates that it's just a normal file, and `d` indicates it's a
+directory. Then, there are three characters describing what the owning user can
+do. The first of these three indicates whether or not the
+owner can read the file (`r` if they can, `-` if they can't), the second (`r` or
+`-`) indicates whether they can write to the file, and the third (`x` or `-`)
+indicates whether or not they can execute this file. We have another group of
+the same three that describe permissions for members of the owning group, and a
+final group of three that describe permissions for everyone else.
+
+If this feels a bit abstract, we can actually see these permissions working with
+a few commands. You can create a file with the `touch` command:
+
+```shell
+$ touch perm-example.txt
+```
+
+> This command has no output. One of the recurring themes in older interfaces
+> like this is that if they don't complain, you're probably safe to assume that
+> nothing's gone wrong.
+
+Now, look at how `perm-example.txt` has been created:
+
+```shell
+$ ls -l perm-example.txt
+-rw-r--r--. 1 atalii users 0 Feb  6 08:29 perm-example.txt
+```
+
+We can see that in this case, the owning user is `atalii`, and the owning group
+is `users`. `atalii` has the `rw-` perms associated: They can read, write, and
+execute the file. Anyone in the users group can read the file but do nothing
```

Comment:
```suggestion
We can see that in this case, the owning user is `atalii`, and the owning group
is `users`. `atalii` has the `rw-` perms associated: They can read and write
the file. Anyone in the users group can read the file but do nothing
```

Reply:
(if you mean to introduce the concept of execution here, there's probably a more correct wording here)

Reply:
note that all other comments here are nitpicks: this is the one that motivated me to request changes here.